src/commands/statusline.ts

```
@@ -64,6 +76,26 @@ export const statuslineCommand = define({
 		// Extract session ID from hook data
 		const sessionId = hookData.session_id;
 
+		// Create semaphore instance for statusline
+		const semaphore = SemaphoreFactory.statusline(ctx.values.refreshInterval);
```

ESLint

[error] Unsafe argument of type `any` assigned to a parameter of type `number | undefined`.

(ts/no-unsafe-argument)